mesonbuild/compilers/cpp.py

```diff
@@ -412,3 +413,35 @@ def get_option_link_args(self, options):
 
     def get_compiler_check_args(self):
         return []
+
+
+class CcrxCPPCompiler(CcrxCompiler, CPPCompiler):
+    def __init__(self, exelist, version, compiler_type, is_cross, exe_wrap=None, **kwargs):
+        CPPCompiler.__init__(self, exelist, version, is_cross, exe_wrap, **kwargs)
+        CcrxCompiler.__init__(self, compiler_type)
+
+    def get_options(self):
+        opts = CPPCompiler.get_options(self)
+        opts.update({'cpp_std': coredata.UserComboOption('cpp_std', 'C++ language standard to use',
+                                                         ['none', 'c++03'], # TODO: maybe supports 'c++11'?
+                                                         'none')})
+        return opts
+
+    def get_option_compile_args(self, options):
+        args = []
+        std = options['cpp_std']
+        if std.value != 'none':
+            args.append('-lang=cpp')
```

Comment:
So setting `-lang=cpp` is sets the standard to c++03 instead of the default (I assume c++98)? that seems a little odd to me.

Reply:
You're probably correct actually. It's not clear based on the compiler documentation which standard it supports.

Reply:
I'm not sure but `-lang=cpp` really looks like "treat these files as C++".

Reply:
what happnes if you run the C++ compiler without -lang=cpp, like on "unit tests/common/2 cpp"? Does that work.

Reply:
Effectively this is what it is, but it also allows for the Embedded C++ standard via this option too.

Then `-lang=` is documented at https://www.renesas.com/sg/en/doc/products/tool/doc/005/r20ut3248ej0102_ccrx.pdf  on page 28

I'm slightly leaning towards removing support for Embedded C++ and just put `-lang=cpp` inside `get_always_args` class to simplify things.

Given that the C++ standard is not clear, what should I put inside `get_options` in that case?

Reply:
I saw that, it's not clear to me what ecpp does. It also wasn't clear if you left -lang=cpp of whether the compiler was smart enough to figure out that you were passing C++ files and not C files.

Reply:
If the `-lang` option is omitted, it will figure it out based on the extension.

Reply:
wat

Reply:
In any case, what is your suggestion? Should remove `ec++` and just have `c++98`? or shall I just remove it altogether and put `-lang=cpp` inside the `get_always_args` func?

Reply:
I'd suggest just always `-lang=cpp`, but that's because I don't really understand the difference between ec++ and c++

Reply:
@jpakkane, opinions?

Other than this I'm happy with this series.